acc_acc_modules/1_annotate_acc_modules.py

```diff
@@ -0,0 +1,373 @@
+                {
+                    "gene id": _id,
+                    "median pairwise dist": median_module_dist,
+                    "min pairwise dist": min_dist,
+                    "max pairwise dist": max_dist,
+                }
+            )
+
+    module_dist = pd.DataFrame(rows)
+    module_dist = module_dist.set_index("gene id")
+
+    return module_dist
+
+
+# -
+
+pao1_module_dist = get_intra_module_dist(pao1_gene_annot, pa_prefix="PA")
+pa14_module_dist = get_intra_module_dist(pa14_gene_annot, pa_prefix="PA14_")
+
+pao1_module_dist.head(10)
+
+pa14_module_dist.head(10)
+
+# Add module distance to gene names
+pao1_gene_annot = pao1_gene_annot.merge(
+    pao1_module_dist, left_index=True, right_index=True, how="left"
+)
+pa14_gene_annot = pa14_gene_annot.merge(
+    pa14_module_dist, left_index=True, right_index=True, how="left"
+)
+
+# ## Add KEGG pathways
+#
+# For each pathway, what genes are contained in it. This information is only available for PAO1.
+
+# +
+pao1_pathway_filename = "https://raw.githubusercontent.com/greenelab/adage/7a4eda39d360b224268921dc1f2c14b32788ab16/Node_interpretation/pseudomonas_KEGG_terms.txt"
+
+pao1_pathways = pd.read_csv(pao1_pathway_filename, sep="\t", index_col=0, header=None)
+# -
+
+pao1_pathways[2] = pao1_pathways[2].str.split(";").apply(set)
+pao1_pathways.index = pao1_pathways.index.str.split(" - ").str[0]
+pao1_pathways.head()
+
+gene_to_pathways_df = pd.DataFrame(
+    index=pao1_gene_module_labels.index, columns=list(pao1_pathways.index)
+)
+
+# %%time
+for gene in gene_to_pathways_df.index:
+    gene_to_pathways_df.loc[gene] = [
+        gene in pao1_pathways.loc[pathway, 2] for pathway in pao1_pathways.index
+    ]
+
+gene_to_pathways_df.head()
+
+# Add gene name to pathway information
+pao1_gene_annot = pao1_gene_annot.merge(
+    gene_to_pathways_df, left_index=True, right_index=True, how="left"
+)
+
+# ## Import and format operon
+
+pao1_operon_filename = paths.PAO1_OPERON
+pa14_operon_filename = paths.PA14_OPERON
+
+pao1_operon = pd.read_csv(pao1_operon_filename, index_col=0, header=0)
+pa14_operon = pd.read_csv(pa14_operon_filename, index_col=0, header=0)
+
+pao1_operon = pao1_operon.set_index("locus_tag")
+pa14_operon = pa14_operon.set_index("locus_tag")
+
+print(pao1_operon.shape)
+pao1_operon.head()
+
+print(pa14_operon.shape)
+pa14_operon.head()
+
+# +
+# There are 247 PAO1 genes with multiple annotations
+# This operon df contains annotations from predicted operons based on DOOR database
+# predictions which make up the majority of the operons) as well as some that
+# are curated (i.e. PseudoCAP)
+# There are some that have multiple PseudoCAP annotations too
+
+# Here we will keep the last PseudoCAP annotations
+# Note: Do we want to discard these annotations all together
+# or will these need to be carefully curated to determine which to keep?
+# We will use the curated annotation here
+pao1_operon = pao1_operon[~pao1_operon.index.duplicated(keep="last")]
```

Comment:
Are you sure the curated annotation will always be the last here?

Reply:
Good point. when i looked at the DB it looked like it was sorted, but I can also just sort before to make sure that this is the case